slack/rtm/client.py

```diff
@@ -352,9 +373,17 @@ async def _connect_and_read(self):
                 client_err.SlackApiError,
                 # TODO: Catch websocket exceptions thrown by aiohttp.
             ) as exception:
-                self._logger.debug(str(exception))
                 await self._dispatch_event(event="error", data=exception)
-                if self.auto_reconnect and not self._stopped:
+                error_code = (
+                    exception.response.get("error", None)
+                    if hasattr(exception, "response")
+                    else None
+                )
+                if (
+                    self.auto_reconnect
+                    and not self._stopped
+                    and error_code != "invalid_auth"  # "invalid_auth" is unrecoverable
```

Comment:
By correcting the behavior of `run_async=False` descried in #530 and #613 , the `RTMClient` started doing exponential retries with unrecoverable errors. This condition is added to prevent it for the cases with invalid tokens.